src/app/course/[courseId]/board/page.tsx

```diff
@@ -0,0 +1,93 @@
+"use client";
+
+import Board from "@components/board";
+import { useOfficeHour } from "@hooks/oh/useOfficeHour";
+import { IdentifiableQuestions } from "@interfaces/type";
+import { Box, Button } from "@mui/material";
+import CheckIcon from "@mui/icons-material/Check";
+import React from "react";
+
+interface PageProps {
+  params: {
+    courseId: string;
+  };
+}
+
+const SELECT_ALL = "All";
+
+const Page = (props: PageProps) => {
+  const { course, questions } = useOfficeHour();
+  const [selectedTopics, setSelectedTopics] = React.useState([SELECT_ALL]);
+
+  const topics = [SELECT_ALL, ...Object.keys(course.tags).sort()];
+
+  const getQuestionsByTopic = (topics: string[]) => {
+    if (topics.length === 0) {
+      return [];
+    } else if (topics.includes(SELECT_ALL)) {
+      return questions;
+    } else {
+      return questions.filter((question) =>
+        question.tags.find((tag) => topics.includes(tag))
+      );
+    }
+  };
+
+  const addOrRemoveTopic = (topic: string) => {
+    setSelectedTopics((topics) => {
+      if (topics.includes(topic)) {
+        return topics.filter((other) => other !== topic);
+      } else {
+        return [...topics, topic];
+      }
+    });
+  };
+
```

Comment:
If the user selects the "All" topic, I think we should deselect the other topics.
And if the user selects some other topic, we can deselect the All topic.